### ISSUE TITLE
Uploading large blobs fail on small VMs due to memory pressure

### ISSUE THREAD

**Issue by LosD** [Azure/azure-xplat-cli#1602]
When uploading large blobs using ```azure storage blob```, the upload often fails on small VMs, probably due to an OOM. In this case, an Amazon EC2 micro instance with 1GB of memory, around 200 MB free accoring to ```top```.

The main issues seems to be that micro instances has no swap space (due to no ephemeral storage), and the node process using more than 200 MB of memory.

#### Prerequisites:
Running  on small machine with no swap.
Logged on to azure

#### Steps to reproduce:
1. find a large-ish file, ~500 MB should do it.
2. Use ```azure storage blob upload -q -a <domain> --container <container> -k <key> file

#### Actual result:
node process uses all available memory and upload often ends with a "Killed" message after 64-128 MB uploaded.

#### Expected result:
Upload completed

#### Workaround:
Split file into multiple chunks, none larger than ~50 MB. It still consumes all available memory, but the OOM is gone.

#### Note:
I guess the answer could be "just get some swap", but using 200MB+ for a simple command-line upload seems excessive, and on VMs with no local storage it this is not really a viable option.

#### Versions.
node.js: 0.10.36
azure: 0.8.14
Azure-Storage: 0.4.1

#### Log:
Here is the last operation of an upload that fails with -v:
```
[2015-02-27T18:11:25.932Z]debug : Operation 19 start to run
Percentage: 15.4% (68.00MB/441.47MB) Average Speed: 13.60MB/S Elapsed Time: 00:00:05 [2015-02-27T18:11:26.676Z]debug : FINAL REQUEST OPTIONS:
{ url: 'https://*************.blob.core.windows.net:443/backup/test.tgz?comp=block&blockid=MDU1ZTg0OTctMDAwMDE4',
method: 'PUT',
headers:
{ 'content-length': 4194304,
'x-ms-client-request-id': '0b303ac0-beac-11e4-b817-67ec3783582a',
'user-agent': 'Azure-Storage/0.4.1 (NODE-VERSION v0.10.36; Linux 3.14.27-25.47.amzn1.x86_64)',
'x-ms-version': '2014-02-14',
'x-ms-date': 'Fri, 27 Feb 2015 18:11:26 GMT',
accept: 'application/atom+xml,application/xml',
'Accept-Charset': 'UTF-8',
'content-type': '',
authorization: 'SharedKey *************:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' },
encoding: undefined }
[2015-02-27T18:11:26.941Z]debug : Add operation 20 into batch operation createBlockFromText.
[2015-02-27T18:11:26.942Z]debug : Operation 20 start to run
[2015-02-27T18:11:31.463Z]debug : FINAL REQUEST OPTIONS:
{ url: 'https://*************.blob.core.windows.net:443/backup/test.tgz?comp=block&blockid=MDU1ZTg0OTctMDAwMDE5',
method: 'PUT',
headers:
{ 'content-length': 4194304,
'x-ms-client-request-id': '0bca57e0-beac-11e4-b817-67ec3783582a',
'user-agent': 'Azure-Storage/0.4.1 (NODE-VERSION v0.10.36; Linux 3.14.27-25.47.amzn1.x86_64)',
'x-ms-version': '2014-02-14',
'x-ms-date': 'Fri, 27 Feb 2015 18:11:28 GMT',
accept: 'application/atom+xml,application/xml',
'Accept-Charset': 'UTF-8',
'content-type': '',
authorization: 'SharedKey *************:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' },
encoding: undefined }
Percentage: 15.4% (68.00MB/441.47MB) Average Speed: 2.06MB/S Elapsed Time: 00:00:33 [2015-02-27T18:11:55.083Z]debug : Add operation 21 into batch operation createBlockFromText.
[2015-02-27T18:11:55.592Z]debug : Operation 21 start to run
Killed
```

**Comment by squillace** [comment]
perfect! Thanks; it'll help everyone who comes after you. Glad it's not a blocker for you.

**Comment by yaxia** [comment]
@LosD Thanks for raising the issue and your workaround. We will fix this issue in an upcoming release of our module in such a way that not consuming up the memory.

**Comment by LosD** [comment]
Cool, thanks! :) 
-- 
Med venlig hilsen 
Dennis Du Krøger

**Comment by yaxia** [comment]
It should be fixed since v0.9.8.

**Comment by LosD** [comment]
Thanks! ☺
